src/CoreArbiterServer.cc

```diff
@@ -1017,11 +1016,28 @@ CoreArbiterServer::distributeCores()
         size_t numCoresToMakeManaged =
             numAssignedCores - managedCores.size();
         LOG(NOTICE, "Making %lu cores managed", numCoresToMakeManaged);
-        managedCores.insert(managedCores.end(),
-                              unmanagedCores.end() - numCoresToMakeManaged,
-                              unmanagedCores.end());
-        unmanagedCores.erase(unmanagedCores.end() - numCoresToMakeManaged,
-                              unmanagedCores.end());
+        // Ensure hypertwin of managed core not managed unless unavoidable.
+        for (int i = 0; i < (int) numCoresToMakeManaged; i++) {
+
+            int coreIdToManageIndex = 0;
+            for (int j = 0; j < (int) unmanagedCores.size(); j++) {
+                bool match = false;
```

Comment:
"Match" is not a great variable name; I can't tell from the code what it means, other than "random boolean variable".